tests/test_interop_scenarios.rs

```diff
@@ -0,0 +1,262 @@
+    println!("Testing ciphersuite {:?}", ciphersuite_name);
+    let ciphersuite = Config::ciphersuite(ciphersuite_name).unwrap();
+
+    let number_of_clients = 2;
+    let setup = ManagedTestSetup::new(default_managed_group_config(), number_of_clients);
+    setup.create_clients();
+
+    // Create a group with two members. Includes the process of adding Bob.
+    let group_id = setup
+        .create_random_group(2, ciphersuite)
+        .expect("Error while trying to create group.");
+    let mut groups = setup.groups.borrow_mut();
+    let group = groups.get_mut(&group_id).unwrap();
+
+    let (_, alice_id) = group.members.first().unwrap().clone();
+
+    // Let Alice create an update with a self-generated KeyPackageBundle.
+    setup
+        .self_update(ActionType::Commit, group, &alice_id, None)
+        .expect("Error self-updating.");
+
+    // Check that group members agree on a group state.
+    setup.check_group_states(group);
+});
+
+/// # Remove
+/// A:    Create group
+/// B->A: KeyPackage
+/// C->A: KeyPackage
+/// A->B: Welcome
+/// A->C: Welcome
+/// A->B: Remove(B), Commit
+/// ***:  Verify group state
+ctest!(remove {
+    let ciphersuite_name = CiphersuiteName::try_from(_ciphersuite_code).unwrap();
+    println!("Testing ciphersuite {:?}", ciphersuite_name);
+    let ciphersuite = Config::ciphersuite(ciphersuite_name).unwrap();
+
+    let number_of_clients = 2;
+    let setup = ManagedTestSetup::new(default_managed_group_config(), number_of_clients);
+    setup.create_clients();
+
+    // Create a group with two members. Includes the process of adding Bob.
+    let group_id = setup
+        .create_random_group(2, ciphersuite)
+        .expect("Error while trying to create group.");
+    let mut groups = setup.groups.borrow_mut();
+    let group = groups.get_mut(&group_id).unwrap();
+
+    let (_, alice_id) = group.members.first().unwrap().clone();
+    let (_, bob_id) = group.members.last().unwrap().clone();
+
+    // Have alice remove Bob.
+    setup
+        .remove_clients(ActionType::Commit, group, &alice_id, vec![bob_id])
+        .expect("Error removing Bob from the group.");
+
+    // Check that group members agree on a group state.
+    setup.check_group_states(group);
+});
+
+// TODO #141, #284: The external PSK, resumption and re-init tests should go here.
+
+/// # Large Group, Full Lifecycle
+/// * Create group
+/// * Group creator adds the first M members
+/// * Until group size reaches N members, a randomly-chosen group member adds a
+///   new member
+/// * All members update
+/// * While the group size is >1, a randomly-chosen group member removes a
+///   randomly-chosen other group member
+ctest!(large_group_lifecycle {
+    let ciphersuite_name = CiphersuiteName::try_from(_ciphersuite_code).unwrap();
+    println!("Testing ciphersuite {:?}", ciphersuite_name);
+    let ciphersuite = Config::ciphersuite(ciphersuite_name).unwrap();
+
+    // "Large" is 20 for now.
+    let number_of_clients = 20;
```

Comment:
If i make this 200 it fails quickly. Something appears to be broken.

Reply:
What seems to fail is the `from_utf8` call in the call back function announcing who has been added by whom. It seems like it which can't handle the group and client ids. I'll fix that. But other than that, it seems to work fine .

And by fine, I mean that it's been running on my machine for a while now. But at least it hasn't crashed :-)